mne/io/curry/curry.py

```diff
@@ -222,8 +273,20 @@ def _read_curry_info(curry_paths):
                 ch['coord_frame'] = FIFF.FIFFV_COORD_DEVICE
             all_chans.append(ch)
 
+    ch_count = len(all_chans)
+    assert (ch_count == curry_params.n_chans)  # ensure that we have assembled
+    # the same number of channels as declared in the info (.DAP) file in the
+    # DATA_PARAMETERS section. Git issue #8391
+
+    # sort the channels to assure they are in the order that matches how
+    # recorded in the datafile.  In general they most likely are already in
+    # the correct order, but if the channel index in the data file was
+    # explicitly declared we might as well use it.
+    all_chans = sorted(all_chans, key=lambda ch: ch['ch_idx'])
+
     ch_names = [chan["ch_name"] for chan in all_chans]
     info = create_info(ch_names, curry_params.sfreq)
+    info['meas_date'] = curry_params.dt_start          # for Git issue #8398
     _make_trans_dig(curry_paths, info, curry_dev_dev_t)
 
     for ind, ch_dict in enumerate(info["chs"]):
```

Comment:
`info['chs'][ii]` should only have a fixed set of entries. I'll push a commit that ensures that `info['chs']` does not have any extraneous entries. In the meantime this should fix it:
```suggestion
    for ind, ch_dict in enumerate(info["chs"]):
        all_chans[ind].pop('ch_idx')
```